examples/nirf/rfsa-rfsg-NR-analysis-only-modacc.py

```diff
@@ -299,15 +304,19 @@ def check_for_warning(response, instrument):
         )
     )
     # Once commit is called, the recommended acquisition settings can be queried
-    response = rfmxclient.GetAttributeF64(
+    # For these updates, the proto request (NR's GetAttributeF64)
+    # must match the RequestMessage type (come from nirfmxnr_types).
+    # But for these, since you're getting an INSTR attribute,
+    # I think you should use an instrclient's GetAttribute... RPC calls.
+    response = instrclient.GetAttributeF64(
         nirfmxinstr_types.GetAttributeF64Request(

```

Comment:
Hi @reckenro , if I use instrclient, do I need to use instr types? Sometimes the variables I need are in the specific standard (NR, WCDMA, etc) but I am confused that they are used in a instrclient way.
In this case, I think you are right, they come from instr but in general how do I know? following the proto file?

Reply:
For the most part, yes. So in the nirfmxinstr.proto the [`GetAttirbuteF64` rpc method](https://github.com/ni/grpc-device/blob/main/generated/nirfmxinstr/nirfmxinstr.proto#L46) expects a `GetAttributeF64Request` from the same proto file [here](https://github.com/ni/grpc-device/blob/main/generated/nirfmxinstr/nirfmxinstr.proto#L720). So the rpc method I think always aligns it's Request and Response messages with ones from the same proto file.

And on that Request message for the attribute_id it expects an `NiRFmxInstrAttribute`: https://github.com/ni/grpc-device/blob/main/generated/nirfmxinstr/nirfmxinstr.proto#L723

I think the one thing for these RFmx client's is that they can share the Session used between them if I remember correctly, but most other messages used will come from the same .proto file.

Reply:
And for the most part I reference the .proto files. I think there are ways to get set up with a better python programming experience with typehints and such but I haven't set that up personally.